sdk/go/pulumi-language-go/main.go

```diff
@@ -46,36 +47,47 @@ import (
 	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
 )
 
-func findProgram(binary string) (*exec.Cmd, error) {
-	// we default to execution via `go run`
-	// the user can explicitly opt in to using a binary executable by specifying
-	// runtime.options.binary in the Pulumi.yaml
-	if binary != "" {
-		program, err := executable.FindExecutable(binary)
-		if err != nil {
-			return nil, errors.Wrap(err, "expected to find prebuilt executable")
-		}
-		return exec.Command(program), nil
+func compileProgramCwd(buildID string) (string, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get current working directory")
+	}
+
+	goFileSearchPattern := filepath.Join(cwd, "*.go")
+	if matches, err := filepath.Glob(goFileSearchPattern); err != nil || len(matches) == 0 {
+		return "", errors.Errorf("Failed to find go files for 'go build' matching %s", goFileSearchPattern)
 	}
 
-	// Fall back to 'go run' style executions
-	logging.V(5).Infof("No prebuilt executable specified, attempting invocation via 'go run'")
-	program, err := executable.FindExecutable("go")
 	if err != nil {
-		return nil, errors.Wrap(err, "problem executing program (could not run language executor)")
+		return "", errors.Wrap(err, "failed to compile")
 	}
 
-	cwd, err := os.Getwd()
+	filePrefix := fmt.Sprintf("pulumi-go.%s", buildID)
+	if runtime.GOOS == "windows" {
+		filePrefix = fmt.Sprintf("pulumi-go.%s.exe", buildID)
+	}
+
+	f, err := os.CreateTemp("", filePrefix+".*")
```

Comment:
This will not do what you want, it will be /tmp/boo.exe.352141780 not /tmp/boo.352141780.exe on Windows